### DISCUSSION TITLE
Documentation for IO class

### DISCUSSION THREAD

**Discussion by stc1988** [Moddable-OpenSource/moddable#625] [General]
[io.md](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/io/io.md)  is something older to [TC53 spec](https://github.com/EcmaTC53/spec/blob/master/docs/tc53.md).

For example, namespace prefix is `builtin`  instead of `embedded:io`. 
Do you hava any plans to update this?

**Comment by phoddie** [comment]
@stc1988, thank you for bringing this up. We do intend to update the io.md document. While it remains largely accurate, there have been changes to the TC53 specification since it was written. Here's a partial list:

- Use of `embedded:` prefix for module specifiers
- Eliminate subclassing of IO classes to provide callbacks
- Addition of `target` property to constructor options object 
- Property name changes (I2C changed `scl` to `clock`, etc

A comprehensive update to the document is a significant effort. Before doing that, I'd prefer to make the code changes to synchronize the implementations with the specifications (they are close, but not complete yet).

In the short term, I do have some pending changes to io.md that resolves several of the divergences. I'll get that committed.

If you notice other places the need to be updated, please note them here. Thanks!

**Comment by stc1988** [comment]
Thanks for the reply and updating documents.

There is no description about `SMBus` class and methods of `SMBus` class has some differences between [implementations](https://github.com/Moddable-OpenSource/moddable/blob/public/modules/io/i2c/smbus.js) and  [TC53 spec](https://github.com/EcmaTC53/spec/blob/master/docs/tc53.md#system-management-bus-smbus).
